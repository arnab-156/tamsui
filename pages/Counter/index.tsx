import React from 'react';
import { Link } from 'react-router-dom';

export default function Counter(): React.ReactElement {
  return (
    <main>
      <h1>React Counter</h1>
      <Link to="/">Back to home</Link>
    </main>
  );
}
