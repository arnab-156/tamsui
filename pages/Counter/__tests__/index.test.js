import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Counter from '../index';

function RouterWrappedCounter() {
  return (
    <MemoryRouter>
      <Counter />
    </MemoryRouter>
  );
}

describe('Counter Component', () => {
  test('matches snapshot', () => {
    const tree = renderer.create(<RouterWrappedCounter />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('contains the counter text', () => {
    render(<RouterWrappedCounter />);

    expect(screen.getByRole('heading')).toHaveTextContent('React Counter');
  });
});
