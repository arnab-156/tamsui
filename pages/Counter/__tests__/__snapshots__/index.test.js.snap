// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component matches snapshot 1`] = `
<main>
  <h1>
    React Counter
  </h1>
  <a
    href="/"
    onClick={[Function]}
  >
    Back to home
  </a>
</main>
`;
